Yosemite/YosemiteTests/SnapshotsProvider/FetchResultSnapshotsProviderTests.swift

```diff
@@ -0,0 +1,239 @@
+import XCTest
+import TestKit
+import CoreData
+import Combine
+
+@testable import Yosemite
+import Storage
+
+@available(iOS 13.0, *)
+final class FetchResultSnapshotsProviderTests: XCTestCase {
+
+    private var storageManager: MockupStorageManager!
+
+    private var cancellables = Set<AnyCancellable>()
+
+    var viewStorage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = MockupStorageManager()
+    }
+
+    override func tearDown() {
+        storageManager = nil
+        cancellables.forEach {
+            $0.cancel()
+        }
+        cancellables.removeAll()
+        super.tearDown()
+    }
+
+    func test_objectWithID_returns_the_expected_immutable_object() throws {
+        // Given
+        let insertedAccount = insertAccount(displayName: "Reina Feil", username: "reinafeil")
+
+        viewStorage.saveIfNeeded()
```

Comment:
just curious, any reasons `viewStorage.saveIfNeeded()` is only called in two of the test functions in this class where some models are also inserted?